libcomposefs/lcfs-writer-erofs.c

```diff
@@ -863,11 +865,44 @@ static int write_erofs_shared_xattrs(struct lcfs_ctx_s *ctx)
 	return 0;
 }
 
+static char *str_join(const char *a, const char *b)
+{
+	size_t a_len = strlen(a);
+	size_t b_len = strlen(b);
+	char *res = malloc(a_len + b_len + 1);
+	if (res) {
+		memcpy(res, a, a_len);
+		memcpy(res + a_len, b, b_len + 1);
+	}
+	return res;
+}
+
 static int add_overlayfs_xattrs(struct lcfs_node_s *node)
 {
+	int type = node->inode.st_mode & S_IFMT;
 	int ret;
 
-	if ((node->inode.st_mode & S_IFMT) == S_IFREG && node->inode.st_size > 0) {
+	/* First escape all existing "trusted.overlay.*" xattrs */
+	for (size_t i = 0; i < lcfs_node_get_n_xattr(node); i++) {
+		const char *name = lcfs_node_get_xattr_name(node, i);
+
+		if (str_has_prefix(name, "trusted.overlay.")) {
+			cleanup_free char *renamed =
+				str_join("trusted.overlay.overlay.",
```

Comment:
Is it really worth carrying `str_join` over just `asprintf ("trusted.overlay.overlay.%s", name + strlen("trusted.overlay."))` ?